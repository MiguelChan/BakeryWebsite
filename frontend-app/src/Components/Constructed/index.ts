export * from './NavigationBar';
export * from './SupplierDetailView';
export * from './ViewSuppliers';
export * from './CreateSupplierView';
export * from './EditSupplierView';