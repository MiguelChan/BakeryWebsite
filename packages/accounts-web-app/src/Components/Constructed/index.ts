export * from './ViewAccountsDashboard/ViewAccountsDashboard';
export * from './EditableAccountForm/EditableAccountForm';
