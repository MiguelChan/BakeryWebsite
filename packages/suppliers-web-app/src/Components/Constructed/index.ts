export * from './SupplierDetailView';
export * from './ViewSuppliers';
export * from './CreateSupplierView';
export * from './EditSupplierView';