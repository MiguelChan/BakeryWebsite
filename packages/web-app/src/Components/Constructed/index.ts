export * from './NavigationDrawer';
export * from './SupplierDetailView';
export * from './ViewSuppliers';
export * from './CreateSupplierView';
export * from './EditSupplierView';