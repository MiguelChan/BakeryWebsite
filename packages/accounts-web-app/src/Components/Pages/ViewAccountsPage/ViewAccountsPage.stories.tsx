import React from 'react';
import {
  Story,
  Meta,
} from '@storybook/react';
import { AccountType } from '@mgl/shared-components';
import { ViewAccountsPage } from './ViewAccountsPage';
import { UseGetAccountsState } from '../../../Hooks';
import {
  AccountsAppContext,
  ApplicationContext,
} from '../../../Context';
import { ViewAccountsDashboard } from '../../Constructed';

export default {
  title: 'Components/Pages/ViewAccountsPage',
  component: ViewAccountsPage,
} as Meta;

interface Context {
  appContext: ApplicationContext;
}

const Template: Story<Context> = ({ appContext }) => (
  <AccountsAppContext.Provider value={appContext}>
    <ViewAccountsPage />
  </AccountsAppContext.Provider>
);

export const EmptyTable = Template.bind({});
EmptyTable.args = {
  appContext: {
    useGetAccounts: (): UseGetAccountsState => ({
      accounts: [],
      isLoading: false,
      errorMessage: undefined,
    }),
  },
};

export const LoadingTable = Template.bind({});
LoadingTable.args = {
  appContext: {
    useGetAccounts: (): UseGetAccountsState => ({
      accounts: [],
      isLoading: true,
    }),
  },
};

export const WithData = Template.bind({});
WithData.args = {
  appContext: {
    useGetAccounts: (): UseGetAccountsState => ({
      accounts: [
        {
          accountType: AccountType.Entry,
          id: 'SomeId',
          subAccounts: [],
          title: 'Some Title',
        },
      ],
      isLoading: false,
    }),
  },
};
