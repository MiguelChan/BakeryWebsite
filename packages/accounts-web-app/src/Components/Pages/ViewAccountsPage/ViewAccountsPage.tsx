import React from 'react';
import {
  Link as RouterLink,
} from 'react-router-dom';
import {
  Button,
  useTheme,
} from '@mui/material';
import { ViewAccountsDashboard } from '../../Constructed';
import { SimpleAppTemplate } from '../../Templates';

/**
 * The ViewAccountsPage component.
 *
 * @returns .
 */
export const ViewAccountsPage: React.FunctionComponent = () => {
  const currentTheme = useTheme();

  const renderBody = (): React.ReactElement => (
    <>
      <ViewAccountsDashboard />
      <Button
        component={RouterLink as any}
        to='/accounts/create'
        variant='contained'
        color='primary'
        sx={{
          marginTop: currentTheme.spacing(2),
        }}
      >
        Crear nueva cuenta
      </Button>
    </>
  );

  return (
    <SimpleAppTemplate
      renderBody={renderBody}
    />
  );
};
