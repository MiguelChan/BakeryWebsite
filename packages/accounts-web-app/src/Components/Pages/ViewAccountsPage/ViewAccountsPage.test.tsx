import React from 'react';
import {
  RenderResult,
  render,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import {
  AccountsAppContext,
  ApplicationContext,
} from '../../../Context';
import { ViewAccountsPage } from './ViewAccountsPage';
import { ViewAccountsDashboard } from '../../Constructed';
import { EmptyTable } from './ViewAccountsPage.stories';

describe.skip('ViewAccountsPage', () => {
  interface Context {
    appContext: ApplicationContext;
  }

  const setupComponent = (props: Context): RenderResult => {
    const Component = (
      <AccountsAppContext.Provider value={props.appContext} >
        <MemoryRouter>
          <ViewAccountsPage />
        </MemoryRouter>
      </AccountsAppContext.Provider >
    );

    return render(Component);
  };

  it('Should display an EmptyTable', () => {
    const props = EmptyTable.args as Context;

    const result = setupComponent(props);

    expect(result.container).toMatchSnapshot();
  });
});
