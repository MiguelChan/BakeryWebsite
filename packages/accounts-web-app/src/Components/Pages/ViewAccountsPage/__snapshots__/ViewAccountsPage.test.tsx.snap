// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAccountsPage Should display an EmptyTable 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
  >
    <table
      class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Id
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Nombre de cuenta
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Tipo de Cuenta
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      />
    </table>
    <a
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-lll1vm-MuiButtonBase-root-MuiButton-root"
      href="/accounts/create"
      tabindex="0"
    >
      Crear nueva cuenta
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;

exports[`ViewAccountsPage Should display some Data 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
  >
    <table
      class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Id
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Nombre de cuenta
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Tipo de Cuenta
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            SomeId
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Some Title
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Entry
          </td>
        </tr>
      </tbody>
    </table>
    <a
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-lll1vm-MuiButtonBase-root-MuiButton-root"
      href="/accounts/create"
      tabindex="0"
    >
      Crear nueva cuenta
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
